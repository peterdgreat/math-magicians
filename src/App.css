* {
  padding: 0;
  margin: 0;
}

.main-container {
  margin: 0 auto;
  height: 100vh;
}
