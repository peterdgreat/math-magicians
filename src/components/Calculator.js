import { useState } from 'react';
import Button from './Button';
import './calculator.css';
import calculate from '../logic/calculate';

export default function Calculator() {
  const initialState = {
    total: 0,
    next: 0,
    operation: null,
  };
  const [state, setstate] = useState(initialState);
  const handleClick = (value) => {
    setstate((state) => calculate(state, value));
  };

  const { total, next, operation } = state;
  return (
    <div className="h-bg">
      <header>
        <div className="d-flex justify-content-end px-2">
          <h2>
            {total}
            {operation}
            {next}
          </h2>
        </div>
      </header>
      <section className="d-flex">

        <Button classes="box btn" value="AC" handleClick={handleClick} />
        <Button classes="box btn" value="+/-" handleClick={handleClick} />
        <Button classes="box btn" value="%" handleClick={handleClick} />
        <Button classes="s-box btn" value="&divide;" handleClick={handleClick} />

      </section>
      <section className="d-flex">

        <Button classes="box btn" value="7" handleClick={handleClick} />
        <Button classes="box btn" value="8" handleClick={handleClick} />
        <Button classes="box btn" value="9" handleClick={handleClick} />
        <Button classes="s-box btn" value="x" handleClick={handleClick} />

      </section>
      <section className="d-flex">

        <Button classes="box btn" value="4" handleClick={handleClick} />
        <Button classes="box btn" value="5" handleClick={handleClick} />
        <Button classes="box btn" value="6" handleClick={handleClick} />
        <Button classes="s-box btn" value="-" handleClick={handleClick} />

      </section>
      <section className="d-flex">

        <Button classes="box btn" value="1" handleClick={handleClick} />
        <Button classes="box btn" value="2" handleClick={handleClick} />
        <Button classes="box btn" value="3" handleClick={handleClick} />
        <Button classes="s-box btn" value="+" handleClick={handleClick} />

      </section>
      <section className="d-flex">
        <Button classes="b-box btn" value="0" handleClick={handleClick} />
        <Button classes="box btn" value="." handleClick={handleClick} />
        <Button classes="s-box btn" value="=" handleClick={handleClick} />

      </section>
    </div>
  );
}
