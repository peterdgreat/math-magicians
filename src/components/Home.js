import React from 'react';

export default function Home() {
  return (
    <div className="container">
      <header>
        <h2> Welcome To Our Page</h2>
      </header>
      <section>

        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,
          doloremque.
        </p>
      </section>
    </div>
  );
}
