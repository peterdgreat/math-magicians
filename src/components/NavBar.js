import React from 'react';
import { NavLink } from 'react-router-dom';

export default function NavBar() {
  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <div className="container-fluid">
        <NavLink className="nav-link" to="/">Math Magician</NavLink>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse  d-flex justify-content-end navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item bdr">
              <NavLink className="nav-link " to="/">Home</NavLink>
            </li>
            <li className="nav-item bdr">
              <NavLink className="nav-link" to="/calculator">Calculator</NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/quote">Quote</NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}
