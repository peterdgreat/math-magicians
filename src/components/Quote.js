import React from 'react';

export default function Quote() {
  return (
    <div>
      quote
    </div>
  );
}
