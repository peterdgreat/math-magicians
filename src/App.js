/* eslint-disable react/prefer-stateless-function */
import './App.css';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import Calculator from './components/Calculator';
import NavBar from './components/NavBar';
import Home from './components/Home';
import Quote from './components/Quote';

export default function App() {
  return (
    <BrowserRouter>

      <NavBar />
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/calculator">

          <div className="main-container col-8 d-flex flex-column justify-content-center">
            <Calculator />
          </div>
        </Route>
        <Route path="/quote">
          <Quote />
        </Route>
      </Switch>
    </BrowserRouter>
  );
}
