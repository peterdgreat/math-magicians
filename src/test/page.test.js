import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';
import NavBar from '../components/NavBar';
import App from '../App';

it('renders without crashing', () => {
  render(<Home />);
});

it('renders without crashing', () => {
  render(<Quote />);
});
it('renders without crashing', () => {
  render(<Calculator />);
});
it('renders without crashing', () => {
  render(
    <Router>
      <NavBar />
    </Router>,
  );
});

describe('Testing Container page with jest snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
