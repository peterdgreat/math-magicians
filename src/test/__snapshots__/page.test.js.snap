// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Container page with jest snapshot renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Math Magician
      </a>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse  d-flex-md justify-content-end navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item bdr"
          >
            <a
              aria-current="page"
              className="nav-link  active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item bdr"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/calculator"
              onClick={[Function]}
            >
              Calculator
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/quote"
              onClick={[Function]}
            >
              Quote
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <header>
      <h2>
         Welcome To Our Page
      </h2>
    </header>
    <section>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque.
      </p>
    </section>
  </div>,
]
`;

exports[`Testing Container page with jest snapshot renders correctly 2`] = `
<div
  className="container"
>
  <header>
    <h2>
       Welcome To Our Page
    </h2>
  </header>
  <section>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, doloremque.
    </p>
  </section>
</div>
`;

exports[`Testing Container page with jest snapshot renders correctly 3`] = `
<div
  className="d-flex flex-column justify-content-center align-items-center quote"
>
  <section
    className="col-6 "
  >
    <p
      className="text-center"
    >
      -
    </p>
    <button
      className="btn btn-primary "
      onClick={[Function]}
      type="button"
    >
      Click to get more quotes
    </button>
  </section>
</div>
`;

exports[`Testing Container page with jest snapshot renders correctly 4`] = `
<div
  className="h-bg"
>
  <header>
    <div
      className="d-flex justify-content-end px-2"
    >
      <h2>
        0
        0
      </h2>
    </div>
  </header>
  <section
    className="d-flex"
  >
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="s-box btn"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </section>
  <section
    className="d-flex"
  >
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="s-box btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </section>
  <section
    className="d-flex"
  >
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="s-box btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </section>
  <section
    className="d-flex"
  >
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="s-box btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </section>
  <section
    className="d-flex"
  >
    <button
      className="b-box btn"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="box btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="s-box btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </section>
</div>
`;

exports[`Testing Container page with jest snapshot renders correctly 5`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container-fluid"
  >
    <a
      aria-current="page"
      className="nav-link active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Math Magician
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse  d-flex-md justify-content-end navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item bdr"
        >
          <a
            aria-current="page"
            className="nav-link  active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item bdr"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Testing Container page with jest snapshot renders correctly 6`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;
